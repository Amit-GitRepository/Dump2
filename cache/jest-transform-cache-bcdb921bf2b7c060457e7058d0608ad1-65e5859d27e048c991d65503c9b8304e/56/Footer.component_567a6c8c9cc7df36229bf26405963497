Object.defineProperty(exports,"__esModule",{value:true});var cov_k18sl0xy2=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/Footer.component.js',hash='1babd266327d1a8cbc2367b0fd74963f1e2083bb',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/Footer.component.js',statementMap:{'0':{start:{line:12,column:39},end:{line:17,column:3}},'1':{start:{line:13,column:39},end:{line:13,column:49}},'2':{start:{line:14,column:4},end:{line:16,column:5}},'3':{start:{line:15,column:6},end:{line:15,column:37}},'4':{start:{line:20,column:4},end:{line:20,column:37}},'5':{start:{line:24,column:45},end:{line:24,column:55}},'6':{start:{line:25,column:4},end:{line:44,column:6}},'7':{start:{line:48,column:0},end:{line:51,column:2}},'8':{start:{line:53,column:0},end:{line:59,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:19},end:{line:12,column:20}},loc:{start:{line:12,column:39},end:{line:17,column:3}},line:12},'1':{name:'(anonymous_1)',decl:{start:{line:12,column:39},end:{line:12,column:40}},loc:{start:{line:12,column:45},end:{line:17,column:3}},line:12},'2':{name:'(anonymous_2)',decl:{start:{line:19,column:22},end:{line:19,column:23}},loc:{start:{line:19,column:28},end:{line:21,column:3}},line:19},'3':{name:'(anonymous_3)',decl:{start:{line:23,column:2},end:{line:23,column:3}},loc:{start:{line:23,column:12},end:{line:45,column:3}},line:23}},branchMap:{'0':{loc:{start:{line:14,column:4},end:{line:16,column:5}},type:'if',locations:[{start:{line:14,column:4},end:{line:16,column:5}},{start:{line:14,column:4},end:{line:16,column:5}}],line:14},'1':{loc:{start:{line:30,column:39},end:{line:30,column:135}},type:'cond-expr',locations:[{start:{line:30,column:59},end:{line:30,column:108}},{start:{line:30,column:111},end:{line:30,column:135}}],line:30},'2':{loc:{start:{line:32,column:39},end:{line:32,column:135}},type:'cond-expr',locations:[{start:{line:32,column:59},end:{line:32,column:108}},{start:{line:32,column:111},end:{line:32,column:135}}],line:32},'3':{loc:{start:{line:39,column:21},end:{line:39,column:67}},type:'cond-expr',locations:[{start:{line:39,column:34},end:{line:39,column:42}},{start:{line:39,column:45},end:{line:39,column:67}}],line:39},'4':{loc:{start:{line:40,column:18},end:{line:40,column:95}},type:'cond-expr',locations:[{start:{line:40,column:31},end:{line:40,column:61}},{start:{line:40,column:64},end:{line:40,column:95}}],line:40}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0},f:{'0':0,'1':0,'2':0,'3':0},b:{'0':[0,0],'1':[0,0],'2':[0,0],'3':[0,0],'4':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _Button=require('../Shared/Button/Button.component');var _Button2=_interopRequireDefault(_Button);
var _ISText=require('../Shared/ISText/ISText.component');var _ISText2=_interopRequireDefault(_ISText);
var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _Footer=require('./Footer.style');var _Footer2=_interopRequireDefault(_Footer);
var _helper=require('../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

Footer=function(_Component){_inherits(Footer,_Component);function Footer(){var _ref;var _temp,_this,_ret;_classCallCheck(this,Footer);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=Footer.__proto__||Object.getPrototypeOf(Footer)).call.apply(_ref,[this].concat(args))),_this),_this.

selectLanguage=function(languageChange){cov_k18sl0xy2.f[0]++;cov_k18sl0xy2.s[0]++;return function(){cov_k18sl0xy2.f[1]++;var _ref2=(cov_k18sl0xy2.s[1]++,
_this.props),changeLanguage=_ref2.changeLanguage,language=_ref2.language;cov_k18sl0xy2.s[2]++;
if(languageChange!==language){cov_k18sl0xy2.b[0][0]++;cov_k18sl0xy2.s[3]++;
changeLanguage(languageChange);
}else{cov_k18sl0xy2.b[0][1]++;}
};},_this.

navigateToLanding=function(){cov_k18sl0xy2.f[2]++;cov_k18sl0xy2.s[4]++;
_this.props.goToScreen('Landing');
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(Footer,[{key:'render',value:function render()

{cov_k18sl0xy2.f[3]++;var _ref3=(cov_k18sl0xy2.s[5]++,
this.props),language=_ref3.language,isLoggedIn=_ref3.isLoggedIn,onLogout=_ref3.onLogout;cov_k18sl0xy2.s[6]++;
return(
_react2.default.createElement(_reactNative.View,{style:_Footer2.default.container},
_react2.default.createElement(_reactNative.View,null,
_react2.default.createElement(_ISText2.default,{style:_Footer2.default.languageChangeText,type:'BOLD'},(0,_helper.translate)('SIDE_MENU__CHANGE_LANGUAGE')),
_react2.default.createElement(_reactNative.View,{style:_Footer2.default.languageSwitch},
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:language==='en'?(cov_k18sl0xy2.b[1][0]++,[_Footer2.default.languageSelection,_Footer2.default.activeLanguage]):(cov_k18sl0xy2.b[1][1]++,_Footer2.default.languageSelection),onPress:this.selectLanguage('en')},'EN'),
_react2.default.createElement(_reactNative.View,{style:_Footer2.default.separator}),
_react2.default.createElement(_ISText2.default,{type:'BOLD',style:language==='th'?(cov_k18sl0xy2.b[2][0]++,[_Footer2.default.languageSelection,_Footer2.default.activeLanguage]):(cov_k18sl0xy2.b[2][1]++,_Footer2.default.languageSelection),onPress:this.selectLanguage('th')},'TH'))),


_react2.default.createElement(_reactNative.View,null,
_react2.default.createElement(_Button2.default,{
style:_Footer2.default.navButton,
textStyle:_Footer2.default.navButtonText,
onPress:isLoggedIn?(cov_k18sl0xy2.b[3][0]++,onLogout):(cov_k18sl0xy2.b[3][1]++,this.navigateToLanding),
text:isLoggedIn?(cov_k18sl0xy2.b[4][0]++,(0,_helper.translate)('SIDE_MENU__LOGOUT')):(cov_k18sl0xy2.b[4][1]++,(0,_helper.translate)('SIDE_MENU__LANDING'))}))));




}}]);return Footer;}(_react.Component);cov_k18sl0xy2.s[7]++;


Footer.defaultProps={
goToScreen:_noop2.default,
changeLanguage:_noop2.default};cov_k18sl0xy2.s[8]++;


Footer.propTypes={
goToScreen:_propTypes2.default.func,
changeLanguage:_propTypes2.default.func,
language:_propTypes2.default.string,
isLoggedIn:_propTypes2.default.bool,
onLogout:_propTypes2.default.func};exports.default=


Footer;