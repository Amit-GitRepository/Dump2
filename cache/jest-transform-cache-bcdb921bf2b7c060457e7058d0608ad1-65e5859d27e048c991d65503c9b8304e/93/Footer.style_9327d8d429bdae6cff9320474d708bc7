Object.defineProperty(exports,"__esModule",{value:true});var cov_v6uiubgl5=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/Footer.style.js',hash='98220dca85049f537bcbc309efdd6d35851e9a7c',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/SideMenu/Footer.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../themes/constants.styles');exports.default=

{
container:{
backgroundColor:_constants.colors.SECONDARY_BG_TEXT_CONTRAST,
paddingHorizontal:15,
paddingVertical:5,
flexDirection:'row',
alignItems:'center',
justifyContent:'space-between'},

navButton:{
paddingVertical:3,
paddingHorizontal:20},

navButtonText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL},

languageChangeText:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_SMALL},

languageSelection:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
fontSize:_constants.fonts.FONT_SIZE_SMALL},

activeLanguage:{
color:_constants.colors.PRIMARY_ACTIONABLE},

languageSwitch:{
flexDirection:'row'},

separator:{
borderWidth:1,
borderColor:_constants.colors.PRIMARY_TEXT_TAB_LABEL,
marginHorizontal:10,
marginVertical:4}};