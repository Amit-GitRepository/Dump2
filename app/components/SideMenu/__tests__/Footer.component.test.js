import Footer from '../Footer.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('Footer component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Footer/>);
    const component = renderer.create(<Footer />).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
});
