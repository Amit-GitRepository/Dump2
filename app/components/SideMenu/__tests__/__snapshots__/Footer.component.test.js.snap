// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 15,
      "paddingVertical": 5,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFFFFF",
          "fontFamily": "SukhumvitSet-Bold",
          "fontSize": 14,
        }
      }
    >
      Change Language
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FFFFFF",
            "fontFamily": "SukhumvitSet-Bold",
            "fontSize": 14,
          }
        }
      >
        EN
      </Text>
      <View
        style={
          Object {
            "borderColor": "#303C46",
            "borderWidth": 1,
            "marginHorizontal": 10,
            "marginVertical": 4,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FFFFFF",
            "fontFamily": "SukhumvitSet-Bold",
            "fontSize": 14,
          }
        }
      >
        TH
      </Text>
    </View>
  </View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F93E60",
          "borderRadius": 50,
          "opacity": 1,
          "padding": 10,
          "paddingHorizontal": 20,
          "paddingVertical": 3,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  },
                  Object {
                    "fontSize": 14,
                  },
                ],
                Object {
                  "fontFamily": "SukhumvitSet-Bold",
                },
              ]
            }
          >
            Landing
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
